shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -421,12 +424,15 @@ - (void)viewDidDisappear:(BOOL)animated {
   TRACE_EVENT0("flutter", "viewDidDisappear");
   [self surfaceUpdated:NO];
   [[_engine.get() lifecycleChannel] sendMessage:@"AppLifecycleState.paused"];
-
+  [_engine.get() setViewController:nil];
   [super viewDidDisappear:animated];
 }
 
 - (void)dealloc {
   [[NSNotificationCenter defaultCenter] removeObserver:self];
+  if (_engineIsOwnedByMe) {
+    [_engine.release() dealloc];
```

Comment:
Hm.  I think you might be right about the dealloc part on the engine.

The cycle is happening because the FlutterViewController is never telling the engine it can be let go of, and so the FVC isn't getting deallocated today.

I'll try to see if I can get a better idea of who else is holding on to the engine...

Reply:
I've modified this so that we now have a `shutdown` method on `FlutterEngine` and can call that.  That allows us to release the shell and shutdown the task runners without explicitly deallocating it.

Reply:
This is better than current state but still has leaks...

Reply:
Still seeing some leaking semantics related stuff, not entirely sure if that's related or should be fixed in a separate PR.